src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java

```diff
@@ -241,6 +248,52 @@ public void initializeFromItineraryAndQueryParams() throws IllegalArgumentExcept
         }
     }
 
+    /**
+     * Updates the mode in the OTP query parameter based on the modes from the itinerary.
+     */
+    private void updateModeInQueryParams() throws URISyntaxException {
+        Set<String> actualModes = itinerary.legs.stream()
+            .map(leg -> leg.mode)
+            .collect(Collectors.toSet());
+
+        // Remove WALK if non-car access modes are present (i.e. {BICYCLE|MICROMOBILITY}[_RENT]).
+        // Removing WALK is necessary for OTP to return certain bicycle+transit itineraries.
+        // Including WALK is necessary for OTP to return certain car+transit itineraries.
+        boolean hasAccessModes = actualModes.stream().anyMatch(mode -> {
+            String mainMode = mode.split("_")[0];
+            return List.of("BICYCLE", "MICROMOBILITY").contains(mainMode);
+        });
+        if (hasAccessModes) {
+            actualModes.remove("WALK");
+        }
+
+        // Replace the "CAR" in the set of modes with the correct CAR query mode (CAR_PARK, CAR_RENT, CAR_HAIL)
+        // (assuming there is only one car leg in an itinerary).
+        Optional<Leg> findCarLeg = itinerary.legs.stream().filter(leg -> "CAR".equals(leg.mode)).findAny();
```

Comment:
Can this be renamed `carLeg`?

Reply:
Updated in https://github.com/ibi-group/otp-middleware/pull/104/commits/1354bae3d3917783f408d667ef115e7435db5cf5.